### ISSUE TITLE
Section: AWS Services / Step Function

### ISSUE THREAD

**Issue by szikszail** [Idea-Pool/aws-testing#24]
Links:
- https://docs.aws.amazon.com/step-functions/latest/dg/create-sample-projects.html
- https://docs.aws.amazon.com/step-functions/latest/dg/sfn-local.html
- https://jira.epam.com/jira/browse/EPMDEBIP-352

Source: https://github.com/Idea-Pool/aws-testing/blob/main/docs/aws-services/step-function.md

Tasks:
- [ ] Introduce steps functions
- [ ] Write about testing types and practices
- [ ] Make sample stack
- [ ] Make tests for the sample stack

Other useful material:
[On the Diverse And Fantastical Shapes of Testing.pdf](https://github.com/Idea-Pool/aws-testing/files/11419945/On.the.Diverse.And.Fantastical.Shapes.of.Testing.pdf)
[Testing AWS Step Functions flows - DEV Community.pdf](https://github.com/Idea-Pool/aws-testing/files/11419946/Testing.AWS.Step.Functions.flows.-.DEV.Community.pdf)
[Testing strategies for Step Functions - Lumigo.pdf](https://github.com/Idea-Pool/aws-testing/files/11419947/Testing.strategies.for.Step.Functions.-.Lumigo.pdf)

**Comment by szikszail** [comment]
Tracking of https://jira.epam.com/jira/browse/EPMDEBIP-352

**Comment by kornel-filep** [comment]
My proposal would be to use 2 (or maybe all 3) from these examples:

- [Fan out a batch job using AWS Step Functions](https://docs.aws.amazon.com/step-functions/latest/dg/sample-batch-fan-out.html)
This seems like a general use-case where we have to fan out a big batch job. As this comes up a lot during development, including an example with this could be interesting and useful for the audience.
- [Dynamically process data with a Map state](https://docs.aws.amazon.com/step-functions/latest/dg/sample-map-state.html)
Data processing and writing to a DynamoDB would increase the scope of this step function testing, but as step functions are used a lot in data pipelines, this might be an interesting scenario to work on as well.
- [Poll for Job Status (Lambda, AWS Batch)](https://docs.aws.amazon.com/step-functions/latest/dg/sample-project-job-poller.html)
And this one is just an easy job poller, which when a timeout happens returns with an error. We have a simple condition check in the step function so might as well include it as a simpler example.

@sorosz89 what do you think?

**Comment by sorosz89** [comment]
Good idea, we can even use a simple one as well: [Make a call to API Gateway](https://docs.aws.amazon.com/step-functions/latest/dg/sample-apigateway-workflow.html)

**Comment by szikszail** [comment]
https://github.com/Idea-Pool/poc-aws-step-function-cdk-testing

**Comment by kornel-filep** [comment]
> https://github.com/Idea-Pool/poc-aws-step-function-cdk-testing

We could use this one as well, however in the scope of the topic we would also like to focus on testing step functions as a solution, with reasonable isolation, instead of pure infrastructure testing.
